src/hotspot/share/runtime/synchronizer.cpp

```diff
@@ -540,8 +555,10 @@ void ObjectSynchronizer::enter_for(Handle obj, BasicLock* lock, JavaThread* lock
   // the locking_thread with respect to the current thread. Currently only used when
   // deoptimizing and re-locking locks. See Deoptimization::relock_objects
   assert(locking_thread == Thread::current() || locking_thread->is_obj_deopt_suspend(), "must be");
-  JavaThread* current = locking_thread;
-  CHECK_THROW_NOSYNC_IMSE(obj);
+  if (obj->klass()->is_inline_klass()) {
+    // JITed code should never have locked an instance of a value class
+    ShouldNotReachHere();
+  }
```

Comment:
You could make this an assert instead.
assert(!EnableValhalla || !obj->klass()->is_inline_klass(), "^ above comment");

Then it's self documenting.  Better than ShouldNotReachHere.